scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/Cli.kt

```diff
@@ -105,10 +109,26 @@ object Cli {
 
     fun configWith(rawArgs: Array<String>): Configuration {
         val fromProperties = ConfigurationProperties.fromResource("local.properties")
+        val fromSystem = getSystemConfig()
         val fromEnv = EnvironmentVariables(prefix = ENV_VAR_PREFIX)
         val fromArgs = parseArguments(rawArgs)
 
-        return fromArgs overriding fromEnv overriding fromProperties
+        return fromArgs overriding fromEnv overriding fromSystem overriding fromProperties
+    }
+
+    private fun getSystemConfig(): Configuration {
+        val dataflowIdPair = this.dataflowReplicaId to getDataflowId()
+        return ConfigurationMap(dataflowIdPair)
+    }
+
+    private fun getDataflowId(): String {
+        return try {
+            InetAddress.getLocalHost().hostName
+        } catch (e: Exception) {
+            val hexCharPool: List<Char> = ('a'..'f') + ('0'..'9')
```

Comment:
nit: is there a package for uuid?

Reply:
When first writing this, I thought that an UUID would be too long for what we need (and slicing out of the UUID provides fewer guarantees due to the constrained format (version/variant fields that are not random).

However, then I went and generated a 50 char hex random string, which is longer than the 32 we would get with a UUID :-). In short, I'll use a UUID v4 here as you propose